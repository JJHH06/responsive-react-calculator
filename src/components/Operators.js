import React, {useEffect, useState} from 'react'

const Operators = ({simbolo, operador}) =>{
    return (
        <div className = "col calculator-operators">
            <p className = "operators-caption">{simbolo}</p>
        </div>
    )
}

export default Operators