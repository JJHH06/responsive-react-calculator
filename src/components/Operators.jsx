/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
// se desactivaron las reglas que no dejan que funcionen los botones
import React from 'react';
import PropTypes from 'prop-types';

const Operators = ({ simbolo, operador, pushOperation }) => {
  const clickOperatorListener = () => {
    pushOperation(operador);
  };

  return (
    <div className="col calculator-operators" onClick={clickOperatorListener}>
      <p className="operators-caption">{simbolo}</p>
    </div>
  );
};

Operators.propTypes = {
  simbolo: PropTypes.string.isRequired,
  operador: PropTypes.string.isRequired,
  pushOperation: PropTypes.func.isRequired,
};

export default Operators;
