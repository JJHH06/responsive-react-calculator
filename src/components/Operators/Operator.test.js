import React from 'react';
import ReactDOM from 'react-dom';
import { isTSAnyKeyword } from '@babel/types';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Operators from './Operators';

afterEach(cleanup);
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Operators />, div);
});

it('renders Operator correctly', () => {
  const { getByTestId } = render(<Operators simbolo="รท" testid="operator/" operador="/" />);
  expect(getByTestId('operator/')).toHaveTextContent('รท');
});

it('renders second Operator correctly', () => {
  const { getByTestId } = render(<Operators simbolo="x" testid="operator*" operador="*" />);
  expect(getByTestId('operator*')).toHaveTextContent('x');
});

it('matches snapshot', () => {
  const tree = renderer.create(<Operators simbolo="x" testid="operator*" operador="*" />).toJSON();
  expect(tree).toMatchSnapshot();
});
