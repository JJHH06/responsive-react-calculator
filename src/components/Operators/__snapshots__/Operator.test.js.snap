// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="col calculator-operators"
  data-testid="operator*"
  onClick={[Function]}
>
  <p
    className="operators-caption"
  >
    x
  </p>
</div>
`;
